infrastructure/dpladm/bin/migrate-site-database-to-incluster-db.mjs

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env zx
+
+import * as crypto from "crypto";
+
+const project = `${process.argv[3]}`;
+if (!project) {
+  throw Error("No 'project' provided");
+}
+
+const environment = `${process.argv[4]}`;
+if (!environment) {
+  throw Error("No 'environment' provided");
+}
+
+let dryRun = `${process.argv[5]}`;
+if (typeof `${process.argv[5]}` === "undefined") {
+  dryRun = false;
+}
+
+const password = crypto.randomBytes(32).toString("base64");
+
+await createDatabaseGrantUserSecret(project, environment, password, dryRun);
+sleep(3000)
+await dumpCurrentDatabaseIntoTmp(project, environment);
+await importDumpIntoInclusterDatabase(project, environment, password);
+await createOverrideVariables(project, environment, password);
+// we need to redeploy the environment for it to take effect ("force" forces yes to prompts)
+await $`lagoon deploy latest --project ${project} --environment ${environment} --force`
+
+
+async function createOverrideVariables(project, environment, password) {
+  const overrideVariables = [
+    {
+      name: "MARIADB_DATABASE_OVERRIDE",
+      value: `database-${project}-${environment}`,
+    },
+    {
+      name: "MARIADB_USERNAME_OVERRIDE",
+      value: `user-database-${project}-${environment}`,
+    },
+    {
+      name: "MARIADB_PASSWORD_OVERRIDE",
+      value: `${password}`,
+    },
+    {
+      name: "MARIADB_HOST_OVERRIDE",
+      value: "mariadb-10-6-01-primary.mariadb-servers.svc.cluster.local",
+    },
+    {
+      name: "MARIADB_PORT_OVERRIDE",
+      value: "3306",
+    }
+  ];
+
+  for (const variable of overrideVariables) {
+    try {
+      await $`lagoon add variable --project ${project} --environment ${environment} --name ${variable.name} --scope global --value ${variable.value}`;
+    } catch (error) {
+      throw Error("failed to create or update variables for BNF and GO secrets", { cause: error });
+    }
+  }
+}
+
+async function importDumpIntoInclusterDatabase(project, environment, password) {
+  echo(`Importing ${project}-${environment} database into incluster database`);
+  await $`kubectl exec -n mariadb-servers mariadb-10-6-01-0 -- bash -c "mariadb -uuser-database-${project}-${environment} -p${password} --database database-${project}-${environment} --verbose < /tmp/dump.sql"`
```

Comment:
Should namespace and server name be configurable here?

Reply:
as mentioned here: https://github.com/danskernesdigitalebibliotek/dpl-platform/pull/668#discussion_r2075197049 the sole task of this piece of script is to enable us to migrate from the Azure manged database to the incluster database. This script contains some pieces that we might reuse when we at some point update the in-cluster database, but I see this as a script made for this point in time. It is maybe worth noting here that we have gone above our estimates on this task already and had it extended, so this task needs to be solved as efficiently as possible, hence why it is not as encompassing as we might have wanted it to be